open_schools_platform/common/filters.py

```diff
@@ -82,29 +96,41 @@ def __init__(self, data: dict = None, queryset: QuerySet = None, **kwargs):
             raise ValidationError(self.errors)
 
     def OR(self, queryset, field_name, value):
-        if type(value) is not str:
-            raise ValidationError(detail="Search field must be str type.")
+        if not or_search_filter_is_valid(value):
+            raise ValidationError(
+                detail="or_search field must be in value:[filter1,filter2,...] format, without spaces after : sign."
+            )
 
-        self.search_value = value
+        self.or_search = value
         return queryset
 
     @property
     def qs(self):
         base_queryset = super().qs.all(force_visibility=self.force_visibility.value)
-
         if self.ORDER_FIELD:
             base_queryset = base_queryset.order_by(self.ORDER_FIELD)
 
-        if not self.search_value:
+        if not self.or_search:
             return base_queryset
 
         query = Q()
-        filters = self.get_filters()
-        for filter in filters.values():
-            if type(filter) is CharFilter and filter.field_name != self.OR_SEARCH_FIELD:
-                query |= Q(**{"{0}__icontains".format(filter.field_name): self.search_value})
 
-        return base_queryset.filter(query)
+        or_search_list = self.or_search.rsplit(":", 1)
+        or_search_value = or_search_list[0]
+        or_search_filters = or_search_list[1].strip("][").split(",")
```

Comment:
make function for this logic that takes string with or_search value and returns  or_search_value and or_search_filters